### PR TITLE
Iterator over all edges in a GraphMap.

### PR DESCRIPTION
**Pull request by contradictioned**
Added an iterator for all edges in a GraphMap.
Added test for this iterator.

Based on the discussion in #17

### CODE REVIEW

File: src/graphmap.rs [hunk 1]
```diff
@@ -327,6 +338,23 @@ impl<'a, N, E> Iterator for Edges<'a, N, E>
     }
 }
 
+pub struct EdgeIter<'a, N, E: 'a> where N: 'a + NodeTrait {
+    pub inner: HashmapIter<'a, (N, N), E>
```
Comment (bluss) [inline src/graphmap.rs]: Ah this should not be `pub`

File: src/graphmap.rs [hunk 2]
```diff
@@ -327,6 +338,23 @@ impl<'a, N, E> Iterator for Edges<'a, N, E>
     }
 }
 
+pub struct EdgeIter<'a, N, E: 'a> where N: 'a + NodeTrait {
```
Comment (bluss) [inline src/graphmap.rs]: The struct should be named after the method that creates it.

**Comment by bluss** [comment]
Thanks! Just fix up some small things and it's ready to go

**Comment by contradictioned** [comment]
Btw: I made the inner iterator `pub` such that it fits the style of `Edges` [some lines above](https://github.com/bluss/petulant-avenger-graphlibrary/blob/master/src/graphmap.rs#L302-L307); maybe these should also be not-pub then.

**Comment by bluss** [comment]
What in the world, that can't be intentional. De-pubbing edges will be put on the todolist for breaking changes.

**Comment by bluss** [comment]
Thank you

**Comment by contradictioned** [comment]
You're welcome :)

**Comment by bluss** [comment]
Published in 0.1.13.

**Pull request merged**